### PR TITLE
fix: fix samples region tags and test flakiness

### LINKED ISSUE #679: Fix exceptions in sample test logs and region tags
The following issues should be addressed about samples and samples tests:

1. Flakiness in `testListLogEntriesSample()` test. It sometimes fails because unexpected log entries (from another tests) are listed and assert validation fail
2. Output of the samples tests run contains `java.lang.RuntimeException` exception:
    ```java
    java.lang.RuntimeException: ManagedChannel allocation site
            at io.grpc.internal.ManagedChannelOrphanWrapper$ManagedChannelReference.<init>(ManagedChannelOrphanWrapper.java:93)
            at io.grpc.internal.ManagedChannelOrphanWrapper.<init>(ManagedChannelOrphanWrapper.java:53)
            at io.grpc.internal.ManagedChannelOrphanWrapper.<init>(ManagedChannelOrphanWrapper.java:44)
            at io.grpc.internal.ManagedChannelImplBuilder.build(ManagedChannelImplBuilder.java:625)
            at io.grpc.internal.AbstractManagedChannelImplBuilder.build(AbstractManagedChannelImplBuilder.java:264)
            at com.google.api.gax.grpc.InstantiatingGrpcChannelProvider.createSingleChannel(InstantiatingGrpcChannelProvider.java:383)
            at com.google.api.gax.grpc.InstantiatingGrpcChannelProvider.access$1900(InstantiatingGrpcChannelProvider.java:82)
            at com.google.api.gax.grpc.InstantiatingGrpcChannelProvider$1.createSingleChannel(InstantiatingGrpcChannelProvider.java:239)
            at com.google.api.gax.grpc.ChannelPool.create(ChannelPool.java:72)
            at com.google.api.gax.grpc.InstantiatingGrpcChannelProvider.createChannel(InstantiatingGrpcChannelProvider.java:249)
            at com.google.api.gax.grpc.InstantiatingGrpcChannelProvider.getTransportChannel(InstantiatingGrpcChannelProvider.java:227)
            at com.google.api.gax.rpc.ClientContext.create(ClientContext.java:205)
            at com.google.api.gax.rpc.ClientContext.create(ClientContext.java:136)
            at com.google.cloud.logging.spi.v2.GrpcLoggingRpc.<init>(GrpcLoggingRpc.java:148)
            at com.google.cloud.logging.LoggingOptions$DefaultLoggingRpcFactory.create(LoggingOptions.java:61)
            at com.google.cloud.logging.LoggingOptions$DefaultLoggingRpcFactory.create(LoggingOptions.java:55)
            at com.google.cloud.ServiceOptions.getRpc(ServiceOptions.java:560)
            at com.google.cloud.logging.LoggingOptions.getLoggingRpcV2(LoggingOptions.java:129)
            at com.google.cloud.logging.LoggingImpl.<init>(LoggingImpl.java:122)
            at com.google.cloud.logging.LoggingOptions$DefaultLoggingFactory.create(LoggingOptions.java:46)
            at com.google.cloud.logging.LoggingOptions$DefaultLoggingFactory.create(LoggingOptions.java:41)
            at com.google.cloud.ServiceOptions.getService(ServiceOptions.java:540)
            at com.example.logging.ListLogEntries.main(ListLogEntries.java:37)
            at com.example.logging.LoggingIT.testWriteLogHttpRequestSample(LoggingIT.java:120)
            at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
            at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
            at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
            at java.lang.reflect.Method.invoke(Method.java:498)
            at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
            at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
            at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
            at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
            at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
            at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
            at java.util.concurrent.FutureTask.run(FutureTask.java:266)
            at java.lang.Thread.run(Thread.java:748)
    ```
3. A sample for the region tag `logging_write_log_entry` does not exist. Another snippet (LogEntryWriteHttpRequest) is tagged with non-existing `logging_write_request_entry` instead of `logging_write_log_entry_advanced`.

### PR DESCRIPTION
**Pull request by minherz**
Set `logging_write_log_entry` and `logging_write_log_entry_advanced` region tags to the correct sample snippets.
Update samples test for the snippets accordingly.
Resolve `java.lang.RuntimeException` exception that appears in the samples tests output.
Modify the samples test to eliminate incorrect output in the faked stdout that leads to failed asserts (in testListLogEntriesSample).

Fixes #679

### CODE REVIEW

File: samples/snippets/src/test/java/com/example/logging/LoggingIT.java [hunk 1]
```diff
@@ -41,16 +43,38 @@
...
-  private Logging logging = LoggingOptions.getDefaultInstance().getService();
+  private static Logging logging;
+
+  private void deleteLog(String logName) throws InterruptedException {
+    int deleteAttempts = 0;
+    int allowedDeleteAttempts = 5;
+    boolean deleted = false;
+    while (!deleted && deleteAttempts < allowedDeleteAttempts) {
+      deleted = logging.deleteLog(logName);
+      deleteAttempts++;
```
Comment (simonz130) [inline samples/snippets/src/test/java/com/example/logging/LoggingIT.java:59]: nit: log deleting log attempts so when we see a failing test, it's easier to understand what happened.

File: samples/snippets/src/test/java/com/example/logging/LogsIT.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+import static com.google.common.truth.Truth.assertThat;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class LogsIT {
```
Comment (simonz130) [inline samples/snippets/src/test/java/com/example/logging/LogsIT.java:27]: document class purpose

**Review by simonz130 — APPROVED** [review]

File: samples/snippets/src/test/java/com/example/logging/LogsIT.java (hunk 2 above)
Comment (minherz) [reply samples/snippets/src/test/java/com/example/logging/LogsIT.java:27]: Lines 64-67 print to STDERR in a case the log is not deleted. Due to sequential nature of the tests, if after few fail attempts the log is deleted, the number of failed attempts is not important.

File: samples/snippets/src/test/java/com/example/logging/LoggingIT.java (hunk 1 above)
Comment (minherz) [reply samples/snippets/src/test/java/com/example/logging/LoggingIT.java:59]: Lines 64-67 print to STDERR in a case the log is not deleted. Due to sequential nature of the tests, if after few fail attempts the log is deleted, the number of failed attempts is not important.

File: samples/snippets/src/test/java/com/example/logging/LogsIT.java (hunk 2 above)
Comment (minherz) [reply samples/snippets/src/test/java/com/example/logging/LogsIT.java:27]: Will do next time I will work on the repository

**Pull request merged**